dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/analytics/AnalyticsSettings.java

```diff
@@ -60,4 +70,66 @@ public String getTableType() {
 
     return EMPTY;
   }
+
+  private boolean isCitusDisabledByConfig() {
+    return dhisConfigurationProvider.isEnabled(CITUS_EXTENSION_DISABLED);
+  }
+
+  public synchronized boolean isCitusEnabled(JdbcTemplate jdbcTemplate) {
+    if (Objects.isNull(citusEnabled)) {
+      if (isCitusDisabledByConfig()) {
+        log.info("Citus extension is disabled in dhis.conf");
+        citusEnabled = false;
+      } else {
+        citusEnabled = isCitusExtensionInstalledAndEnabled(jdbcTemplate);
+      }
+    }
+    return citusEnabled;
+  }
+
+  private boolean isCitusExtensionInstalledAndEnabled(JdbcTemplate jdbcTemplate) {
+    List<PgExtension> installedExtensions = getPostgresExtensions(jdbcTemplate);
+
+    logFoundExtensions(installedExtensions);
+
+    return installedExtensions.stream().anyMatch(this::isCitusEnabled);
+  }
+
+  private void logFoundExtensions(List<PgExtension> installedExtensions) {
+    if (!installedExtensions.isEmpty()) {
+      log.info(
+          "Found citus extensions in the database: {}",
+          StringUtils.join(installedExtensions, ", "));
+      return;
+    }
+    log.info(
+        "Citus extension software is not installed in the database. You need to install it first, depending on the OS: https://docs.citusdata.com/en/latest/installation/multi_node.html");
+  }
+
+  private List<PgExtension> getPostgresExtensions(JdbcTemplate jdbcTemplate) {
```

Comment:
Is this going to query globally? Or is it able to identify if the extension is created for a specific DB (based on the JDBC URL)?